src/connector/src/sink/mqtt.rs

```diff
@@ -244,15 +248,45 @@ struct MqttSinkPayloadWriter {
     retain: bool,
 }
 
-impl FormattedSink for MqttSinkPayloadWriter {
-    type K = Vec<u8>;
-    type V = Vec<u8>;
+impl MqttSinkPayloadWriter {}
+
+impl MqttSinkPayloadWriter {
+    async fn write_chunk<F: SinkFormatter>(
+        &mut self,
+        chunk: StreamChunk,
+        formatter: &F,
+    ) -> Result<()>
+    where
+        F::K: SerTo<Vec<u8>>,
+        F::V: SerTo<Vec<u8>> + QueryField,
+    {
+        for r in formatter.format_chunk(&chunk) {
+            let (_, event_object) = r?;
+            let topic = {
+                if let Some(name) = &self.config.topic_field {
+                    match &event_object {
+                        Some(event_object) => {
+                            event_object.get_field(name).context("column not found")?
```

Comment:
In which case do we meet the error? Is it a common practice to forward the message to `self.config.common.topic`?
I think throwing an error here is not a good idea, where RisingWave will keep trying to deliver the message to a non-exist topic, keep getting the error here, and then stuck the stream. cc @wenym1

Reply:
I think it's better to be explicit about the error and let the user know about these mistakes, is there a way to stop the sink in case of an error in the execution?

Reply:
Agree. I think it's better to be explicit about the error. 

But I think we can still improve the user experience by adding a new option to be the default topic, so that user can specify the behavior if the the topic field is not set. If the default topic options is not set while the topic field is not set in the row, we can then  return with error.